server/api/autocomplete.go

```diff
@@ -0,0 +1,36 @@
+package api
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/store"
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/utils"
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/utils/bot"
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/utils/httputils"
+)
+
+func (api *api) autocompleteUsers(w http.ResponseWriter, r *http.Request) {
+	// TODO: This doesn't seem like proper authorization. Anyone can send a header to an http
+	// request impersonating the user (?)
+	mattermostUserID := r.Header.Get("X-Mattermost-User")
```

Comment:
I think this should be `Mattermost-User-Id` instead. Where does `X-Mattermost-User` come from?

```suggestion
	mattermostUserID := r.Header.Get("Mattermost-User-Id")
```

Check out these links for more info on why this is a trusted approach. If that header is defined, this is an authenticated request for that user. If it's not defined, it's not an authenticated request.

- https://developers.mattermost.com/integrate/plugins/components/server/reference/#Hooks.ServeHTTP
- https://github.com/mattermost/mattermost/blob/93a2c3281ad896a84fc2a3fd6a52708a765a7dd9/server/channels/app/plugin_requests.go#L153
- https://github.com/mattermost/mattermost/blob/93a2c3281ad896a84fc2a3fd6a52708a765a7dd9/server/channels/app/plugin_requests.go#L209

Reply:
Honestly I got it from here: https://developers.mattermost.com/integrate/plugins/components/server/best-practices/#how-can-plugins-make-sure-http-requests-are-authentic but I don't know why I implemented it with another name 🤔